### PR TITLE
fix for ipfs hash calculation for big files

### LINKED ISSUE #2604: Problem with IPFS Hashing Binary files
**Describe the bug**
Running `/script/generate_ipfs_hashes.py` after a binary file inside a package is changed (e.g. `packages/fetchai/contracts/erc1155/build/erc1155.wasm`) returns the following error:

```bash
Processing package erc1155 of type contract
Traceback (most recent call last):
  File "./scripts/generate_ipfs_hashes.py", line 430, in update_hashes
    update_fingerprint(configuration_obj, client)
  File "./scripts/generate_ipfs_hashes.py", line 342, in update_fingerprint
    configuration.directory, configuration.fingerprint_ignore_patterns, client
  File "./scripts/generate_ipfs_hashes.py", line 323, in compute_fingerprint
    assert_hash_consistency(fingerprint, package_path, client)
  File "./scripts/generate_ipfs_hashes.py", line 280, in assert_hash_consistency
    ), "WARNING, hashes don't match for: {}".format(path)
AssertionError: WARNING, hashes don't match for: packages/fetchai/contracts/erc1155/build/erc1155.wasm
```

**To Reproduce**
1. Change `packages/fetchai/contracts/erc1155/build/erc1155.wasm` with a new binary file.
2. Run `./script/generate_ipfs_hashes.py`
3. See error.

**Expected behavior**
Hashes should be updated without error.

### PR DESCRIPTION
**Pull request by solarw**
## Proposed changes

fix for ipfs hash calculation for big files

## Fixes

#2604

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes and CI passes too
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: aea/helpers/ipfs/base.py [hunk 1]
```diff
@@ -65,9 +64,17 @@ def _read(file_path: str) -> bytes:
...
+def chunks(data: Sized, size: int) -> Generator:
+    """Yield successivesize chunks from data."""
+    for i in range(0, len(data), size):
+        yield data[i : i + size]  # type: ignore
+
+
 class IPFSHashOnly:
     """A helper class which allows construction of an IPFS hash without interacting with an IPFS daemon."""
 
+    DEFAULT_CHUNK_SIZE = 262144
```
Comment (DavidMinarsch) [inline aea/helpers/ipfs/base.py:76]: Where is this coming from? Please add link to relevant documentation on IPFS side

File: aea/helpers/ipfs/base.py [hunk 2]
```diff
@@ -110,8 +147,19 @@ def _generate_multihash(pb_data: bytes) -> str:
...
-        sha256_hash = hashlib.sha256(pb_data).hexdigest()
-        multihash_hex = SHA256_ID + LEN_SHA256 + sha256_hash
-        multihash_bytes = codecs.decode(str.encode(multihash_hex), "hex")
+        multihash_bytes = cls._generate_multihash_bytes(pb_data)
         ipfs_hash = base58.b58encode(multihash_bytes)
         return str(ipfs_hash, "utf-8")
+
+
+def fix_dag_encode() -> None:
+    """Fix it to make compatible with ipfs way mof encoding: Links first."""
```
Comment (DavidMinarsch) [inline aea/helpers/ipfs/base.py:156]:
```suggestion
    """Fix it to make compatible with ipfs way of encoding: Links first."""
```

**Review by DavidMinarsch — COMMENTED** [review]
Some of the contract packages currently ignore the binary files in the yaml for hashing. Please remove ignores so we can see it works for them too!

**Comment by solarw** [comment]
> Some of the contract packages currently ignore the binary files in the yaml for hashing. Please remove ignores so we can see it works for them too!

updated

File: aea/helpers/ipfs/utils.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+# ------------------------------------------------------------------------------
+"""This module contains utility methods for ipfs helpers."""
+import contextlib
+import os
+import sys
+from typing import Generator
+
+
+@contextlib.contextmanager
+def _protobuf_python_implementation() -> Generator:
```
Comment (DavidMinarsch) [inline aea/helpers/ipfs/utils.py:27]: Can you explain in the docstring why this is required please

Comment (solarw) [reply aea/helpers/ipfs/utils.py:27]: done

**Review by DavidMinarsch — APPROVED** [review]
LGTM

File: packages/fetchai/contracts/oracle/contract.yaml [hunk 4]
```diff
@@ -8,10 +8,11 @@ aea_version: '>=1.0.0, <2.0.0'
 fingerprint:
   README.md: QmWgKLF6Fe4tESTdhStxYJrEVKP2xSthd2fcMm2LSfkygL
   __init__.py: QmRmf3Q5F6nQMrTEt59suBNw6zfgTVQX68E7CAuKmCnhAM
+  build/FetchOracle.json: QmfEMai1yxPtWoshFahBk2EyVHd9Mo8pSp1SAE83rRvQgH
```
Comment (5A11) [inline packages/fetchai/contracts/oracle/contract.yaml:11]: Do the same for `packages/fetchai/contracts/erc1155/contract.yaml`

Comment (solarw) [reply packages/fetchai/contracts/oracle/contract.yaml:11]: done

**Review by 5A11 — APPROVED** [review]

**Pull request merged**